api/security_firewall.go

```diff
@@ -104,6 +104,22 @@ func (api *API) ReadFirewallSettings(instanceID int) ([]map[string]interface{},
 	return nil, fmt.Errorf("ReadFirewallSettings failed, status: %v, message: %s", response.StatusCode, failed)
 }
 
+func (api *API) ReadFirewallRule(instanceID int, ip string) (map[string]interface{}, error) {
+	data, err := api.ReadFirewallSettings(instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dt := range data {
+		for k, v := range dt {
+			if k == "ip" && v == ip {
+				return dt, nil
+			}
```

Comment:
What does `dt` stand for? Maybe we can use some more descriptive name for the variable

Reply:
True `rule` in this context.